### PR TITLE
Bump LTS. Fix name shadowing.

### PR DESCRIPTION
**Pull request by lwm**

### CODE REVIEW

File: src/Stack/Config.hs [hunk 1]
```diff
@@ -75,7 +75,7 @@ import           Network.HTTP.Client (parseUrlThrow)
 import           Network.HTTP.Download (download)
 import           Network.HTTP.Simple (httpJSON, getResponseBody)
 import           Options.Applicative (Parser, strOption, long, help)
-import           Path
+import           Path hiding (fileExtension)
```
Comment (borsboom) [inline src/Stack/Config.hs]: Looks like we're getting some warnings on [Travis](https://travis-ci.org/commercialhaskell/stack/jobs/189485298):

```
src/Stack/Config.hs:78:1: Warning:
    Module ‘Path’ does not export ‘fileExtension’
```

Comment (lwm) [inline src/Stack/Config.hs]: Hmmmm .... strange! Must have read this one wrong. Fixup coming ... :+1:

**Comment by lwm** [comment]
```
--  While building package store-0.3 using:
      /home/travis/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_1.24.0.0_ghc-8.0.1 --builddir=.stack-work/dist/x86_64-linux/Cabal-1.24.0.0 build --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure (-9) (THIS MAY INDICATE OUT OF MEMORY)
    Logs have been written to: /home/travis/build/commercialhaskell/stack/.stack-work/logs/store-0.3.log
```

:|

Will push again to fire that build ...

**Comment by lwm** [comment]
Well, not sure what to do with that OOM. Maybe we don't need that LTS bump then ;)

**Comment by lwm** [comment]
Let's see what Travis thinks now ...

**Pull request closed without merging**